src/DataManagement/Saving/AlternativeSavingCalculator.cs

```diff
@@ -6,49 +6,90 @@
-            decimal startingBalance, SavingType savingType = SavingType.Regular)
+        public CalculationResults GetSuggestedExpensesOffers(IEntryList entryList, IGoal goal, decimal startingBalance, SavingType savingType = SavingType.Regular)
         {
-            entrySuggestions ??= new List<ISavingSuggestion>();
-
+            var entrySuggestions = new List<ISavingSuggestion>();
+            var monthlySuggestions = new List<SavingSuggestionByImportance>();
+            
             if (entryList is null)
             {
-                return false;
+                return new CalculationResults(entrySuggestions, monthlySuggestions, 0);              
             }
 
             var enumCount = Enum.GetValues(typeof(Importance)).Length;
+ 
+            var sumsOfAmountByImportanceAdjusted = new decimal[enumCount];
+            var sumsOfAmountByImportanceDefault = new decimal[enumCount];
+            var averagesOfAmountByImportanceAdjusted = new decimal[enumCount];
+            var averagesOfAmountByImportanceDefault = new decimal[enumCount];
+            var entryAmounts = new int[enumCount];
 
             for (var i = enumCount; i > (int)Importance.Necessary; i--)
             {
-                var expenses = entryList.GetBy((Importance)i - 1);
+                var expenses = entryList.GetBy((Importance)i);
                 var ratio = enumCount - i;
                 foreach (var entry in expenses)
                 {
-                    var amountAfterSaving = savingType switch
+                    var amountAfterSaving = 0M;
+                    switch (savingType)
                     {
-                        SavingType.Minimal => entry.Amount * SavingRatio * ratio * MinimalSavingValue,
-                        SavingType.Regular => entry.Amount * SavingRatio * ratio * RegularSavingValue,
-                        SavingType.Maximal => entry.Amount * SavingRatio * ratio * MaximalSavingValue,
-                        _ => throw new ArgumentOutOfRangeException()
+                        case SavingType.Minimal:
+                            if (entry.Amount * ratio * MinimalSavingValue < entry.Amount)
+                            {
+                                amountAfterSaving = entry.Amount * ratio * MinimalSavingValue;
+                            }
+                            else
+                            {
+                                amountAfterSaving = entry.Amount;
+                            }
+                            break;
+                        case SavingType.Regular:
+                            amountAfterSaving = entry.Amount * ratio * RegularSavingValue;
+                            break;
+                        case SavingType.Maximal:
+                            amountAfterSaving = entry.Amount * ratio * MaximalSavingValue;                            
+                            break;
```

Comment:
I think there should still be default case which throws the exceptions, just in case :D

Reply:
Hmm, well, can't hurt I guess :D

Reply:
Nice that you've implemented that